src/datepicker.jsx

```diff
@@ -188,7 +188,7 @@ var DatePicker = React.createClass({
       return
     }
 
-    if (!isSameDay(this.props.selected, changedDate)) {
+    if (!isSameDay(this.props.selected, changedDate) || this.props.disableDateAutoCorrection) {
```

Comment:
And the journey continues...

While doing some research on why one of the unit tests was failing in NUI Widgets, it was found that typing an invalid date into the input was no longer calling the `onChange` function (from props) since the `isSameDay` function was always returning true because of the invalid dates (null == null).  Although an argument could be made that this is the correct behavior, the current DatePicker in NUI Widgets is relying on this to be called (even when the date may be null) in order to prepare data for the callback functions (`onChange`, `onBlur` and `onFocus`).

A refactoring within the NUI Widget DatePicker would result in a breaking change so it was easier to make this change.  If refactoring was desired, an analysis would need to be made how consumers are using the data passed in the callback functions (it is currently passing the formatted date value, the raw value from the input and a boolean of whether the date is valid).  It _seems_ like consumers should use the now-exposed `onChangeRaw` function to monitor the raw input value and `onChange` for monitoring changes to the date.